### ISSUE TITLE
search.from variable not respected over the course of searches 

### ISSUE THREAD

**Issue by deeso** [radare/radare2#1476]
This bug is based on some observations, and it may require some additional debugging or work to figure out the mechanics of the issue.  I found that radare will not always respect the ```search.from``` variable.  When the variable is first set, radare will respect the variable, but subsequent searches (after some seeks maybe?) do not seem to use the value.  

For example, in order to search the entire file (all sections and maps), the command, ```e search.from=0```, is performed.  Radare will use this address to search from initially.  However, after seeking around the file, another search is performed.  Rather than using the ```search.from``` address, radare starts from the current position (e.g. address) in the file.

**Comment by Maijin** [comment]
```
s 0x1000 ; e search.in =raw ; e search.from=0x0 ; e search.to=0x100 ; /m ; s
or
s 0x1000 ; e search.in =file ; e search.from=0x0 ; e search.to=0x100 ; /m ; s
```
This should search from the offset 0x0 to 0x100 for magic value right ? Looks like it doesn't because seeking offset is 0x1000 + the seek offset is moved after the search which is not the case for example when performing a /x .

**Comment by Maijin** [comment]
ping @radare

**Comment by radare** [comment]
so /m doesnt respects search.to?

**Comment by Maijin** [comment]
Two issues:

* `s 0 ; e search.from = 0 ; /m` -> Works
* `s 1000 ; e search.from = 0 ; /m` -> Doesn't work

----

* `s 0x1000 ; /x 9090cd80 ; s` -> 0x00001000 OK
* `s 0x1000 ; /m ; s` -> 0x100005574 WTF

**Comment by radare** [comment]
See search.in and set it to raw

**Comment by Maijin** [comment]
Doesn't help at all :(

**Comment by radare** [comment]
Do we have test cases for this?

**Comment by Maijin** [comment]
https://github.com/radare/radare2-regressions/commit/e349e932989ec0cc375cc7a57881b47803aba265

**Comment by jroimartin** [comment]
Should be fixed by #4973
